src/sagemaker/processing.py

```diff
@@ -1737,12 +1739,29 @@ def _pack_and_upload_code(
             "runproc.sh",
         )
         script = estimator.uploaded_code.script_name
-        s3_runproc_sh = S3Uploader.upload_string_as_file_body(
-            self._generate_framework_script(script),
-            desired_s3_uri=entrypoint_s3_uri,
-            kms_key=kms_key,
-            sagemaker_session=self.sagemaker_session,
-        )
+
+        # If we are leveraging a pipeline session with optimized s3 artifact paths,
+        # we need to hash and upload the runproc.sh file to a separate location.
+        if _pipeline_config and _pipeline_config.pipeline_name:
```

Comment:
Nit: may be good to extract to a helper function according to [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle)